apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs

```diff
@@ -66,6 +66,41 @@ public abstract partial class StorageClient : IDisposable
         /// </summary>
         public virtual EncryptionKey EncryptionKey { get { throw new NotImplementedException(); } }
 
+        /// <summary>
+        /// A URL signer built base on this client, that is using the same credential as this client, as well
+        /// as defaulting to this client's URI scheme and host. If the credential used by this client is not compatible
+        /// with <see cref="UrlSigner"/> this will be null. See <see cref="UrlSigner"/>'s documentation for more information
+        /// on compatible credentials.
+        /// </summary>
+        /// <remarks>
+        /// Because credentials used by this client may changed, this property will return a new <see cref="UrlSigner"/> instance
```

Comment:
I think it would be better to make this a method (`CreateUrlSigner()`) in that case. It just makes it really clear - and it's unlikely to be significantly harder to use that way.

I'd argue that when it's a method, we should allow any exceptions to propagate instead of returning null, too. (Or maybe have a `TryCreateUrlSigner` method that returns null.)

Reply:
Yes, I debated over this, settled on a property, but I wasn't convinced. I'm adding the method then, CreateUrlSigner and letting exceptions bobble up.